utilityscripts/tiffs/tiffconvert.py

```diff
@@ -0,0 +1,201 @@
+		'-a_srs', 'EPSG:4326',
+		'-co', 'COMPRESS=DEFLATE',
+		'-co', 'PREDICTOR=1',
+		'-of', 'GTiff',
+		'-q'
+	])
+
+	return convert_result
+
+# -------------------------------------------------------------------
+def searchtiffs(startingdir, convert=False, removeBackup=False):
+
+	# confirm GDAL
+	if not checkforgdal():
+		msg('Could not find GDAL. Plz install it', 5)
+
+	if not startingdir:
+		msg('Specify a directory on the command line to search inside of', 5)
+
+	# confirm starting dir
+	if not os.access(startingdir, os.R_OK):
+		msg('Could not access starting directory ( ' + startingdir + ' ).', 5)
+
+	# loop into starting dir
+	for dir, subdirs, files in os.walk(startingdir):
+
+		for file in files:
+
+			filewithpath = os.path.join(dir, file)
+
+			# is this a tiff file?
+			if file[-4:].lower() != '.tif':
+				# not a tif file
+				msg('' + filewithpath + ' does not have .tif extension', 1)
+				continue
+
+			msg('examining ' + filewithpath, 1)
+
+			fileinfo = gdalinfo(filewithpath)
+
+			file_okay = True
+
+			if fileinfo['compression'] != 'DEFLATE':
+				msg('' + filewithpath + ' compression is not DEFLATE', 1)
+				file_okay = False
+
+			if file_okay and fileinfo['srs'] != 'EPSG:4326':
+				msg('' + filewithpath + ' SRS is not EPSG:4326', 1)
+				file_okay = False
+
+			if not file_okay:
+
+				if convert:
+					msg('conversion needed: ' + file, 2)
+
+					copied_original = filewithpath + '.original-pre-conversion'
+
+					if not careful_copy(filewithpath, copied_original):
+						msg('could not copy original -- not processing ' + file, 4)
+						continue
+
+					gdalconvert(copied_original, filewithpath)
+
+					if (removeBackup):
+						# check first
+						newfileinfo = gdalinfo(filewithpath)
+						if newfileinfo['compression'] == 'DEFLATE' and newfileinfo['srs'] == 'EPSG:4326':
+							# .. then we're okay to remove the backup
+							os.remove(copied_original)
+						else:
+							msg('Conversion is suspect, leaving backup: ' + copied_original, 4)
+
+				else:
+					# not converting, so message is required 
+					print('conversion needed: ' + filewithpath)
+
+# -------------------------------------------------------------------
+# -------------------------------------------------------------------
+# -------------------------------------------------------------------
+# commands
+# -------------------------------------------------------------------
+
+def accept_command():
+	parser = argparse.ArgumentParser(add_help=False,
+			description='Recursively examine / convert tiffs in a given directory.',
+	)
+	parser.add_argument('command', metavar='command', default='help', nargs='?', 
+			help='action to take: help (the default), detect, convert, replace.',
+			choices=['help', 'detect', 'convert', 'replace']
+	)
+	parser.add_argument('targetdir', metavar='targetdir', nargs='?', 
+			help="Target directory within which the action will happen."
+	)
+	parser.add_argument('--verbose', dest='verbose', action='store_true')
```

Comment:
With the `logging` module, you can use `logger.setLevel` (in combination with a nargs options for verbosity; eg https://stackoverflow.com/a/8335918/1048705)